Learn-Information-Security-by-Building-a-Contact-Form/server100.js

```diff
@@ -0,0 +1,25 @@
+const express = require("express");
+const app = express();
+const helmet = require("helmet");
+// https://helmetjs.github.io/ - for more information on all packages and uses.
+
+const helmetConfig = {
+  noCache: false,
+  contentSecurityPolicy: {
+    directives: {
+      defaultSrc: ["'self'"],
+      scriptSrc: ["'self'"],
+      styleSrc: ["'self'"]
+    }
+  }
+};
+
+app.use(helmet(helmetConfig));
+
+// With this Content Security Policy only allows content from the defined domains.
```

Comment:
It might be good to have learners actually add a CDN to one of the arrays. You could use Bootstrap as an example and have them include a different CDN:

// With this Content Security Policy in place, only content from defined domains are allowed.
// If you wanted to include Bootstrap, for example, you would add `'maxcdn.bootstrapcdn.com'` to the `styleSrc` array.
// For example:
// `styleSrc: ["'self'", 'maxcdn.bootstrapcdn.com']`
// ...

Then you can prompt the learner to include something like `'fonts.google.com'` in `fontsSrc` so they can use Google Fonts later.

After that, you could prompt learners to write `app.use(express.urlencoded({ extended: false }));`
`app.use(express.json());` in separate steps.